week4/solutions.js

```diff
@@ -118,34 +116,74 @@ R.compose(
   score => +score
 )(task7_payload.payloadString);
 
-// given:
-// const n = 2
-// produce:
-// [{score: 2}, {score: 1}]
+/* #########################################################################
+ Task 8 :
+      input: const n = 2
+      output: [{score: 2}, {score: 1}] 
+############################################################################ */
+const f = n => (n <= 0 ? false : [R.objOf("score")(n), n - 1]);
+R.unfold(f, 2);
+
+/* #########################################################################
+ Task 9:
+      input: {noise: '/factory/1/noise', temperature: '/factory/1/temperature'}
+      output: {'/factory/1/noise': [], '/factory/1/temperature': []}
+############################################################################ */
+
+R.zipObj(R.values(input), R.repeat([], 2));
 
 // {noise: '/factory/1/noise', temperature: '/factory/1/temperature'}
 // to {'/factory/1/noise': [], '/factory/1/temperature': []}
 
-// given:
-//
-// {showDialog: false, buttonDisabled: false, slides: []}
-// {showDialog: true, buttonDisabled: true, slides: [{id: 1, name: 'programming'}]}
-//
-// produce:
-// {showDialog: false, buttonDisabled: false, slides: [{id: 1, name: 'programming'}]}
+/* #########################################################################
+ Task 10: 
+   input:
+   {showDialog: false, buttonDisabled: false, slides: []}
+   {showDialog: true, buttonDisabled: true, slides: [{id: 1, name: 'programming'}]}
+
+   output:
+   {showDialog: false, buttonDisabled: false, slides: [{id: 1, name: 'programming'}]}
+############################################################################ */
+// skip
 
-// {connecting: true, connected: false}
-// to {connecting: false, connected: true}
+/* #########################################################################
+ Task 11:
+   input: {connecting: true, connected: false}
+   output: {connecting: false, connected: true}
+############################################################################ */
 
-// given:
-// const sensorNames = ['noise', 'temperature']
-// const sensorValues = [20, 30]
-// const airQualitySensorData = {airQuality: 40}
-// produce:
-// {noise: 20, temperature: 30, airQuality: 40}
+R.evolve({
```

Comment:
const input11 = {connecting: true, connected: false}
// to {connecting: false, connected: true}
mapObjIndexed((v, k) => !v)(input11)